fili-core/src/main/java/com/yahoo/bard/webservice/table/resolver/DataSourceConstraint.java

```diff
@@ -54,6 +55,39 @@ public DataSourceConstraint(DataApiRequest dataApiRequest, DruidAggregationQuery
         ).collect(Collectors.toSet()));
     }
 
+    /**
+     * Constructor.
+     *
+     * @param requestDimensions  Dimensions contained in request
+     * @param filterDimensions  Filtered dimensions
+     * @param metricDimensions  Metric related dimensions
+     * @param metricNames  Names of metrics
+     * @param allDimensions  Set of all dimension objects
+     * @param allDimensionNames  Set of all dimension names
+     * @param allColumnNames  Set of all column names
+     * @param apiFilters  Map of dimension to its set of API filters
+     */
+    public DataSourceConstraint(
```

Comment:
@michael-mclawhorn This constructor is a copy-constructor that doesn't go through the same defensive copies that the other constructor does. It's only purpose in life is to support the `with*` method that this PR added, and should not be called from outside this class.